src/main/java/org/terasology/changingBlocks/conditional/ConditionalBlocksSystem.java

```diff
@@ -0,0 +1,227 @@
+import org.terasology.changingBlocks.conditional.components.ChangeBlockEntityDirectedComponent;
+import org.terasology.changingBlocks.conditional.components.ChangeBlockEntityNearbyComponent;
+import org.terasology.changingBlocks.conditional.components.ConditionalBlockChangeComponent;
+import org.terasology.entitySystem.entity.EntityManager;
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.entitySystem.entity.lifecycleEvents.BeforeRemoveComponent;
+import org.terasology.entitySystem.entity.lifecycleEvents.OnAddedComponent;
+import org.terasology.entitySystem.event.ReceiveEvent;
+import org.terasology.entitySystem.systems.BaseComponentSystem;
+import org.terasology.entitySystem.systems.RegisterMode;
+import org.terasology.entitySystem.systems.RegisterSystem;
+import org.terasology.entitySystem.systems.UpdateSubscriberSystem;
+import org.terasology.logic.location.LocationComponent;
+import org.terasology.math.Direction;
+import org.terasology.math.Side;
+import org.terasology.math.geom.Vector3f;
+import org.terasology.math.geom.Vector3i;
+import org.terasology.physics.Physics;
+import org.terasology.physics.StandardCollisionGroup;
+import org.terasology.registry.In;
+import org.terasology.utilities.random.FastRandom;
+import org.terasology.utilities.random.Random;
+import org.terasology.world.OnChangedBlock;
+import org.terasology.world.WorldProvider;
+import org.terasology.world.block.BlockComponent;
+import org.terasology.world.block.BlockManager;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@RegisterSystem(RegisterMode.AUTHORITY)
+public class ConditionalBlocksSystem extends BaseComponentSystem {
+
+    @In
+    private BlockManager blockManager;
+    @In
+    private EntityManager entityManager;
+    @In
+    private WorldProvider worldprovider;
+    @In
+    private Physics physics;
+    @In
+    private Random random;
+
+    private Logger log;
+
+    private List<String> blockTriggers = new ArrayList<>();
+    private List<String> entityTriggers = new ArrayList<>();
+    private Map<String, List<EntityRef>> triggerCollections = new HashMap<>();
+
+    @Override
+    public void initialise() {
+        random = new FastRandom();
+        log = LoggerFactory.getLogger(this.getClass());
+        log.info("Logger created");
+    }
+
+    //TODO: handle Entity-triggered changes, not just Block-triggered changes
+
+    @ReceiveEvent(components = {BlockComponent.class, LocationComponent.class})
+    public void onUpdate(OnChangedBlock event, EntityRef entity) {
+        //if this block is a registered trigger
+        String trigger = event.getNewType().getURI().toString();
+        if (blockTriggers.contains(trigger)) {
+            //get every changer
+            for (EntityRef blockChange : triggerCollections.get(trigger)) {
+                ChangeBlockBlockNearbyComponent bn = blockChange.getComponent(ChangeBlockBlockNearbyComponent.class);
```

Comment:
I might also be a better idea to send a new event with the involved entities and react on that instead of manually checking the components... 🤔 not sure, though.

Reply:
Refactoring this to smaller methods for better readability would still be appreciated.